### DISCUSSION TITLE
Удаленный доступ к базе данных PostgreSQL (TablePlus, Yandex.DataLens)

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#313] [Как это сделать?] [answered]
Необходимо настроить подключение к базе данных PostgreSQL через TablePlus. Настройки на стороне сервера сделаны. В частности открыт удаленный доступ для всех IP адресов. Сообщение об ошибке " password authentication failed for user "postgres". Хотя пароль точный. В чем может быть ошибка ?

**Comment by poluninatm** [comment]
Посмотрите, правильное ли имя базы данных у вас забито при подключении - я несколько раз так накалывалась, оставив предложенное программой.

**Reply by vldekov** [reply]
usename  | usesuper | usecreatedb 
----------+----------+-------------
 postgres | t        | t
 totum    | f        | f
(2 rows)

**Reply by vldekov** [reply]
List of databases
   Name    |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   
-----------+----------+----------+-------------+-------------+-----------------------
 postgres  | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 template0 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
 template1 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
 totum     | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =Tc/postgres         +
           |          |          |             |             | postgres=CTc/postgres+
           |          |          |             |             | totum=CTc/postgres

**Reply by vldekov** [reply]
Я правильно понял, что база данных называется totum, а суперпользователь postgres ?

**Comment by vldekov** [comment]
база данных "totum"

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Вот может это будет полезно: https://ru.docs.totum.online/connect_tableau

[Еще обсуждение про фармат данных в БД](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/313#discussioncomment-4719246)

**Reply by ttm-alexeypolunin** [reply]
Если это стандартная установка то:

база: totum
пользователь: totum
пароль смотреть в: /home/totum/totum-mit/Conf.php
схема: totum

**Reply by vldekov** [reply]
Спасибо! Подключился )

**Comment by vldekov** [comment]
УПС! ) После подключения базы данных PostgreSQL к TablePlus обнаружилась другая проблема. Например, данные типа строка в таблице имеют формат {"v": "тест1"}. А в Yandex.DataLens эти поля определяются как "Неподдерживаемый тип". Подскажите как можно решить эту проблему с форматом данных?

**Reply by ttm-alexeypolunin** [reply]
Тотум использует тип jsonb который является структурой. Поэтому при запросе надо использовать обращение внутрь:

```
select first_field->>'v' as "field 1", second_field->>'v' as "field 2" from totum.tableau where (second_field->>'v')::NUMERIC > 20
```

Ну и в видео выше я показываю на примере Tableau.

Если эта хрень от Яндекса не может выполнять запрос ([оказалось, что может](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/313#discussioncomment-4734963)), то вам нужно сделать view в самой БД, который будет преобразовывать данные в нужные форматы. Я попрошу завтра программистов сделать пример команды создающей view.

**Reply by ttm-alexeypolunin** [reply]
Создание view в БД:

```
create view users__v(id, login, fio, boss_id, roles, add_users, all_connected_users, interface, favorite, is_del) as
SELECT users.id,
       (users.login ->> 'v'::text) AS login,
       (users.fio ->> 'v'::text) AS fio,
       ((users.boss_id ->> 'v'::text))::integer AS boss_id,
       (users.roles ->> 'v'::text) AS roles,
       (users.add_users ->> 'v'::text) AS add_users,
       (users.all_connected_users ->> 'v'::text) AS all_connected_users,
       (users.interface ->> 'v'::text) AS interface,
       (users.favorite ->> 'v'::text) AS favorite,
       users.is_del
FROM users;
```

**Comment by vldekov** [comment]
Ответ Яндекс : "Для того, чтобы в DataLens определился тип как «Строка» в источнике он должен быть типа char, varchar или text. Возможно в вашей БД всё же используется другой тип, что и привело к ошибке."

**Comment by vldekov** [comment]
С преобразованием типа данных разобрались на стороне Yandex.DataLens. Можно прислать пример написания view в самой БД totum?

**Reply by ttm-alexeypolunin** [reply]
Скажите по возможности, что вы сделали в DataLens, что бы я мог отправлять на этот топик людей с аналогичным вопросом. Спс


[Ответ здесь](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/313#discussioncomment-4736866)

**Comment by vldekov** [comment]
В DataLens есть возможность написать запросы в QL-чартах. Всё также как в Вашем примере с Tableau.
